### PR TITLE
ui: run builds (PROJQUAY-6297)

### PR DESCRIPTION
**Pull request by bcaton85**
Adds the ability for users to run builds via the Patternfly UI.

### CODE REVIEW

**Review by arborite-rh — APPROVED** [review]
brushing up my on TS but lgtm

**Review by harishsurf — COMMENTED** [review]
Overall looks good, except for some console errors.
Some minor nits I found (not all are related to this PR). They can be followed up in a separate PR but adding them here for reference:
 1.  When clicking on start new build -> under upload docker file view -> click on start build (without selecting dockerfile) - a failed alert is shown which displays JS  `[object object]`. Also, it would be nice if it’s possible to disable the start build button if a file is not uploaded.
2. The choose robot accounts wizard step in custom build trigger wizard uses a radio button + clear button which differs from other pages which use checkboxes.
3. The trigger name col has excess white space at the end, which truncates subsequent column name titles with a `…`
4.  The text wrapped inside <code>  tags could be styled in red color for easier identification as seen in mocks

File: web/src/routes/RepositoryDetails/Builds/BuildHistoryStartBuildModal.tsx [hunk 1]
```diff
@@ -0,0 +1,118 @@
...
+  Tooltip,
+} from '@patternfly/react-core';
+import {Table, Tbody, Td, Th, Thead, Tr} from '@patternfly/react-table';
+import {useState} from 'react';
+import {RepositoryBuildTrigger} from 'src/resources/BuildResource';
+import BuildTriggerDescription from './BuildTriggerDescription';
+import Conditional from 'src/components/empty/Conditional';
+import DockerfileUploadBuild from './BuildHistoryStartBuildModalUploadDockerfile';
+
+export default function StartBuildModal(props: StartNewBuildModalProps) {
```
[the remark this answers is not in the document]
Comment (bcaton85) [reply web/src/routes/RepositoryDetails/Builds/BuildHistoryStartBuildModal.tsx:18]: fixed

File: web/src/routes/RepositoryDetails/Builds/Builds.tsx [hunk 2]
```diff
@@ -2,18 +2,40 @@ import {Card, PageSection} from '@patternfly/react-core';
 import BuildHistory from './BuildHistory';
 import BuildTriggers from './BuildTriggers';
 import {RepositoryDetails} from 'src/resources/RepositoryResource';
+import {useBuildTriggers} from 'src/hooks/UseBuildTriggers';
+import {LoadingPage} from 'src/components/LoadingPage';
+import RequestError from 'src/components/errors/RequestError';
 
 export default function Builds(props: BuildsProps) {
+  const {triggers, isLoading, isError, error} = useBuildTriggers(
```
[the remark this answers is not in the document]
Comment (bcaton85) [reply web/src/routes/RepositoryDetails/Builds/Builds.tsx:10]: fixed

File: web/src/routes/RepositoryDetails/Builds/BuildHistoryManuallyStartTriggerModal.tsx [hunk 3]
```diff
@@ -0,0 +1,156 @@
...
+      id="manually-start-build-modal"
+      isOpen={props.isOpen}
+      onClose={() => props.onClose()}
+      variant={ModalVariant.medium}
+      actions={[
+        <Button
+          key="start-build"
+          isDisabled={!isValid}
+          onClick={() =>
+            startBuild(trigger?.service === 'custom-git' ? commit : ref)
```
[the remark this answers is not in the document]
Comment (bcaton85) [reply web/src/routes/RepositoryDetails/Builds/BuildHistoryManuallyStartTriggerModal.tsx:81]: I wasn't able to reproduce this. If you let me know the exact steps i can try again.

File: web/src/routes/RepositoryDetails/Builds/BuildHistoryStartBuildModal.tsx [hunk 4]
```diff
@@ -0,0 +1,118 @@
...
+              {props.triggers.map((trigger) => (
+                <Tr key={trigger.id}>
+                  <Td>
+                    <BuildTriggerDescription trigger={trigger} />
+                  </Td>
+                  <Td>{trigger?.config?.branchtag_regex || 'All'}</Td>
+                  <Td>
+                    <Conditional if={trigger.can_invoke}>
+                      <a onClick={() => props.onSelectTrigger(trigger)}>
+                        Run Trigger Now
```
[the remark this answers is not in the document]
Comment (bcaton85) [reply web/src/routes/RepositoryDetails/Builds/BuildHistoryStartBuildModal.tsx:75]: Fixed, i removed the option

**Comment by bcaton85** [comment]
Great review. I'll open up a separate JIRA for the other issues you mentioned.

File: web/src/routes/RepositoryDetails/Builds/BuildHistoryManuallyStartTriggerModal.tsx (hunk 3 above)
[the remark this answers is not in the document]
Comment (harishsurf) [reply web/src/routes/RepositoryDetails/Builds/BuildHistoryManuallyStartTriggerModal.tsx:81]: I also see the earlier warning too when you run the newly created trigger using `Run Trigger Now` kebab option:
```
BuildHistoryManuallyStartTriggerModal.tsx:81 Warning: Can't perform a React state update on an unmounted component. This is a no-op, but it indicates a memory leak in your application. To fix, cancel all subscriptions and asynchronous tasks in a useEffect cleanup function.
    at ManuallyStartTrigger (http://localhost:9000/main.bundle.js:169001:25)
```

[the remark this answers is not in the document]
Comment (bcaton85) [reply web/src/routes/RepositoryDetails/Builds/BuildHistoryManuallyStartTriggerModal.tsx:81]: okay let me try again and see if i can reproduce. I added the descendent warning to the misc JIRA since that isn't introduced by this PR.

File: web/src/routes/RepositoryDetails/Builds/BuildHistoryStartBuildModal.tsx (hunk 4 above)
[the remark this answers is not in the document]
Comment (harishsurf) [reply web/src/routes/RepositoryDetails/Builds/BuildHistoryStartBuildModal.tsx:75]: how about simply greying out the option instead of removing the kebab option so that the user sees it and knows its disabled? We could use the `isDisabled` prop on [DropdownItem](https://www.patternfly.org/components/menus/application-launcher/#dropdownitem)

**Review by harishsurf — COMMENTED** [review]
Thanks for addressing the reviews. Except for browser warning rest of it lgtm!  
We should probably also add a validation to verify file type when uploading Dockerfile (as done in the old UI) - this can be followed up too since it's not related to this PR

File: web/src/routes/RepositoryDetails/Builds/BuildHistoryStartBuildModal.tsx (hunk 4 above)
[the remark this answers is not in the document]
Comment (bcaton85) [reply web/src/routes/RepositoryDetails/Builds/BuildHistoryStartBuildModal.tsx:75]: i agree, that's probably a better UX

File: web/src/routes/RepositoryDetails/Builds/BuildHistoryManuallyStartTriggerModal.tsx (hunk 3 above)
[the remark this answers is not in the document]
Comment (bcaton85) [reply web/src/routes/RepositoryDetails/Builds/BuildHistoryManuallyStartTriggerModal.tsx:81]: I fixed the `<ul> cannot appear as a descendant of <p>` issue. I was able to reproduce the memory leak error but i had to go through the prompt ~12 times. This is an unlikely scenario so I don't think this is a big issue imo.

**Review by harishsurf — APPROVED** [review]
lgtm!  🚀

**Review by harishsurf — APPROVED** [review]
